src/sql/src/func.rs

```diff
@@ -1940,6 +1940,26 @@ pub static PG_CATALOG_BUILTINS: Lazy<BTreeMap<&'static str, Func>> = Lazy::new(|
         "current_database" => Scalar {
             params!() => UnmaterializableFunc::CurrentDatabase, 861;
         },
+        "current_setting" => Scalar {
+            params!(String) => Operation::unary(|_ecx, e| {
+                match e.into_literal_string() {
```

Comment:
I'm worried about this panicing if the argument isn't a string, but I'm not sure how that would ever happen.

Reply:
I guess you won't be able to use this in more complex queries like:

```sql
SELECT name, current_setting(name) FROM (VALUES ('cluster'), ('cluster_replica')) AS R(name);
```
The following works in Postgres:
```sql
SELECT name, current_setting(name) FROM (VALUES ('autovacuum_analyze_scale_factor'), ('autovacuum_analyze_threshold')) AS R(name)
```
I guess if we _really_ need this somebody will complain.

Reply:
cc @umanwizard :point_up: ...

Reply:
> I'm worried about this panicing if the argument isn't a string, but I'm not sure how that would ever happen.

It seems like we follow this same pattern in a bunch of function definitions so I think it should be fine. Plus you have a test that passes a non-string here so hopefully that would catch any panics.

Reply:
We do it in `regexp_extract`, but that is a mistake, and I don't want that pattern to spread any further! Peeking inside values in the SQL planner breaks the notion of what a value is. We need to do the slightly harder thing of allowing arbitrary values, like I mentioned on Slack.

The trick is to have the unmaterializable function return _all_ settings, and then, here, call that function, and plan a `map_get` that looks up the as-yet-unknown user-specified name for the variable in the list of all variables.